cli.js

```diff
@@ -78,19 +79,25 @@ xfs.mkdirp(program.output, async err => {
 
   // If we want to watch for changes do that
   if (program.watchTemplate) {
+    let watcher;
     const watchDir = path.resolve(template);
+    const outputPath = path.resolve(watchDir, program.output);
     // Template name is needed as it is not always a part of the cli commad
     // There is a use case that you run generator from a root of the template with `./` path
     const templateName = require(path.resolve(watchDir,'package.json')).name;
 
-    console.log(`[WATCHER] Watching for changes in the template directory ${magenta(watchDir)} and in the AsyncAPI file ${magenta(asyncapiFile)}`);
-
+    if (isAsyncapiDocLocal) {
+      console.log(`[WATCHER] Watching for changes in the template directory ${magenta(watchDir)} and in the AsyncAPI file ${magenta(asyncapiDocPath)}`);
+      watcher = new Watcher([asyncapiDocPath, watchDir], outputPath);
+    } else {
```

Comment:
This just reminded me this "else" should always be the case. The param is called `watchTemplate` and we're watching the AsyncAPI file too. We should have a different param for watching the AsyncAPI file, like `--watch` (thinking from a user perspective, watching the template is not interesting except for template developers). Should we create a follow-up issue?

Reply:
I'm not sure. With my previous bug fixes I explicitly specified that `--watch-template` is for development only and I was thinking if there might be a use case that someone uses a template with a watch enabled while editing asyncapi file....and I could not imagine someone doing it really, that is what AsyncAPI Hub is for 😄

Reply:
I'll create an issue for it for further discussion, I don't think generator is a tool that should solve a need for nice AsyncAPI document development experience, this is something for AsyncAPI CLI that we do not have yet :D

Reply:
https://github.com/asyncapi/generator/issues/352